lua/lspconfig/util.lua

```diff
@@ -364,10 +377,8 @@ function M.get_other_matching_providers(filetype)
   for _, config in pairs(configs) do
     if not vim.tbl_contains(active_clients_list, config.name) then
       local filetypes = config.filetypes or {}
-      for _, ft in pairs(filetypes) do
-        if ft == filetype then
-          table.insert(other_matching_configs, config)
-        end
+      if M.ft_matches(filetype, filetypes) then
+        table.insert(other_matching_configs, config)
```

Comment:
these changes seem like they might be useful, but `ft_matches` could stay private, it doesn't need to be a new public function.

Reply:
See: concerns raised in https://github.com/neovim/nvim-lspconfig/pull/3145#discussion_r1783179475